### PR TITLE
jellyfin: fix arm{,64} compilation

### PR DESCRIPTION
**Pull request by JCapucho**
###### Description of changes

It seems like one of the nuget dependencies is platform dependant and
since the nuget dependencies were only being generated for x64 the other
architectures were missing their version of the dependency causing the
build to fail on them.

So I split the nuget dependencies file into one for each architecture.

PS: I'm not very familiar with C# and packaging it, so my approach might not be the best

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Pinging maintainers:
@nyanloutre @minijackson @purcell @jojosch

### CODE REVIEW

**Review by jojosch — APPROVED** [review]
LGTM - thank you for fixing.

I don't know if there is a "nicer" way of defining the different platform-specific dependencies. It seems that it [was possible to specify multiple runtimes](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore) in the `dotnet restore` command, but that doesn't seem to work anymore:

image

image

**Comment by onny** [comment]
Would be really great to have this fixed. I guess this is the best solution until it is fixed upstream somehow.

**Review by purcell — APPROVED** [review]
I'd also be keen to see this fixed, and the change broadly looks good to me: just a bit concerned the workaround might linger unnecessarily if this is fixed upstream.

**Comment by rafaelcgs10** [comment]
I confirm this fixes the build process, but I can't start the service.

Here is the relevant part of the log:

```
[2022-06-22 12:38:54.449 +00:00] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind addresses: "[]"
[2022-06-22 12:38:54.449 +00:00] [INF] [1] Jellyfin.Networking.Manager.NetworkManager: Using bind exclusions: "[]"
[2022-06-22 12:38:59.186 +00:00] [INF] [1] Jellyfin.Server.Implementations.JellyfinDbProvider: There are pending EFCore migrations in the database. Applying... (This may take a while, do not stop Jellyfin)
[2022-06-22 12:38:59.542 +00:00] [FTL] [1] Main: Error while starting server.
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'UNIQUE constraint failed: Preferences.UserId, Preferences.Kind'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteNonQuery()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteNonQuery(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migrationCommands, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.Migrate(DatabaseFacade databaseFacade)
   at Jellyfin.Server.Implementations.JellyfinDbProvider..ctor(IServiceProvider serviceProvider, IApplicationPaths appPaths, ILogger`1 logger) in /build/source/Jellyfin.Server.Implementations/JellyfinDbProvider.cs:line 36
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at Emby.Server.Implementations.ApplicationHost.Resolve[T]() in /build/source/Emby.Server.Implementations/ApplicationHost.cs:line 379
   at Emby.Server.Implementations.ApplicationHost.InitializeServices() in /build/source/Emby.Server.Implementations/ApplicationHost.cs:line 659
   at Jellyfin.Server.Program.StartApp(StartupOptions options) in /build/source/Jellyfin.Server/Program.cs:line 195
[2022-06-22 12:38:59.562 +00:00] [INF] [1] Main: Running query planner optimizations in the database... This might take a while
[2022-06-22 12:38:59.599 +00:00] [INF] [1] Jellyfin.Server.Implementations.JellyfinDbProvider: There are pending EFCore migrations in the database. Applying... (This may take a while, do not stop Jellyfin)
[2022-06-22 12:38:59.637 +00:00] [FTL] [1] Main: Unhandled Exception
```

I already had a Jellyfin installation (10.7.7).  I don't know if a DB migration happens in an upgrade like this.

**Comment by JCapucho** [comment]
> I already had a Jellyfin installation (10.7.7). I don't know if a DB migration happens in an upgrade like this.

I also had a 10.7 installation and it upgraded just fine but it seems like other users are also having this problem and there's an open issue about it on the repo [jellyfin#7555](https://github.com/jellyfin/jellyfin/issues/7555)

File: pkgs/servers/jellyfin/update.sh [hunk 1]
```diff
@@ -15,23 +15,36 @@ if [[ "$currentVersion" == "$latestVersion" ]]; then
...
-popd
-rm -r "$src"
+    mkdir ./nuget_tmp.packages
+
+    nugetDepsFile="$realDir/nuget-deps-"$1".nix"
+
+    echo "Updating depencies for $1 at $nugetDepsFile"
+
+    dotnet restore Jellyfin.Server --packages ./nuget_tmp.packages --runtime linux-"$1"
+    nuget-to-nix ./nuget_tmp.packages > "$nugetDepsFile"
```
Comment (mdarocha) [inline pkgs/servers/jellyfin/update.sh:41]: If you move the `nuget-to-nix` invocation *after* all the various `update_nuget_for_arch` invocations, you can streamline the dependencies by generating only one `deps.nix` file. This file will have all the packages restore for the various RIDs, and will be equivalent to calling `dotnet restore` with multiple RIDS, as referenced by https://github.com/NixOS/nixpkgs/pull/177902#pullrequestreview-1011567105

Comment (JCapucho) [reply pkgs/servers/jellyfin/update.sh]: Thanks for the suggestion, it greatly simplifies the changes

**Comment by JCapucho** [comment]
@purcell @jojosch Can I ask you that you re-review the changes?

**Review by jojosch — APPROVED** [review]
LGTM. Runs on x86_64-linux and the Tests pass - I don't have a ARM system to check.

Result of `nixpkgs-review pr 177902` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jellyfin</li>
  </ul>
</details>

**Review by purcell — APPROVED** [review]
LGTM too, thanks for tackling this!

**Review by mdarocha — APPROVED** [review]

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 177902` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jellyfin</li>
  </ul>
</details>

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 177902` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jellyfin</li>
  </ul>
</details>

**Review by felschr — APPROVED** [review]

**Comment by felschr** [comment]
Builds and runs fine on my `aarch64-linux` machine.

**Pull request merged**